web/test.html

```diff
@@ -1,110 +1,210 @@
+    }
+
+  });
+
+  //Funtion to initiailize html elements 
+  function init() {
+    document.myform.url.value = "ws://localhost:8000/"
+    document.myform.inputtext.value = "Hello World!"
+    document.myform.disconnectButton.disabled = true;
+  }
+
+  //Function to make connection to Web-Socket 
+  function doConnect() {
+    websocket = new WebSocket(document.myform.url.value);
+    websocket.onopen = function (evt) { onOpen(evt) };
+    websocket.onclose = function (evt) { onClose(evt) };
+    websocket.onmessage = function (evt) { onMessage(evt) };
+    websocket.onerror = function (evt) { onError(evt) };
+  }
+
+
+  //Function called when connection established with Web-Socket successfully
+  function onOpen(evt) {
+    writeToScreen("connected\n");
+    document.myform.connectButton.disabled = true;
+    document.myform.disconnectButton.disabled = false;
+  }
+
+
+  //Function called when connection is disconnected with Web-Socket successfully
+  function onClose(evt) {
+    writeToScreen("disconnected\n");
+    document.myform.connectButton.disabled = false;
+    document.myform.disconnectButton.disabled = true;
+  }
+
+  //Function called when Web-Socket send response successfully
+  function onMessage(evt) {
+    writeToScreen("response: IMG=" + evt.data + '\n');
+    document.myform.image.src = evt.data;
+  }
+
+  //Function called when Web-Socket sends ERROR message successfully
+  function onError(evt) {
+    writeToScreen('error: ' + evt.data + '\n');
+    websocket.close();
+    document.myform.connectButton.disabled = false;
+    document.myform.disconnectButton.disabled = true;
+
+  }
+
+  //Function called when send request to Web-Socket
+  function doSend(message) {
+    writeToScreen("sent: " + message + '\n');
+    websocket.send(message);
+  }
+
+  //Function called to display message on screen
+  function writeToScreen(message) {
+    document.myform.outputtext.value += message
+    document.myform.outputtext.scrollTop = document.myform.outputtext.scrollHeight;
+  }
+
+
+  //Wrapper function for doSend()
+  function sendText() {
+    doSend(document.myform.inputtext.value);
+  }
+
+  //Function called to clear output text
+  function clearText() {
+    document.myform.outputtext.value = "";
+  }
+
+  //Function called to disconnect with Web-Sockets
+  function doDisconnect() {
+    websocket.close();
+  }
+
+  //Function to find image position on page
+  function findImagePosition(img) {
+    if (typeof (img.offsetParent) != "undefined") {
+      for (var x = 0, y = 0; img; img = img.offsetParent) {
+        x += img.offsetLeft;
+        y += img.offsetTop;
+      }
+      return [x, y];
+    }
+    else {
+      return [img.x, img.y];
+    }
+  }
+
+  //Function to get normalize coordinates of image
+  function GetImageCoordinates(evt) {
+    var x = 0;
+    var y = 0;
+    var imgPos;
+    imgPos = findImagePosition(img);
+    if (!evt) var evt = window.event;
+    if (evt.pageX || evt.pageY) {

```

Comment:
Add a comment explaining why we need to check both pageX and clientX, browser dependent or dependent on what we click on?